projects/packages/my-jetpack/_inc/components/connection-status-card/index.tsx

```diff
@@ -1,326 +1,38 @@
+import { currentUserCan } from '@automattic/jetpack-script-data';
+import { Button } from '@wordpress/components';
 import { useDispatch } from '@wordpress/data';
-import { __, sprintf } from '@wordpress/i18n';
-import { Icon, info, check, lockOutline } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
 import clsx from 'clsx';
-import { useState, useCallback, useMemo } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import { useAllProducts } from '../../data/products/use-all-products';
 import { getMyJetpackWindowInitialState } from '../../data/utils/get-my-jetpack-window-state';
 import getProductSlugsThatRequireUserConnection from '../../data/utils/get-product-slugs-that-require-user-connection';
 import useAnalytics from '../../hooks/use-analytics';
-import useConnectSite from '../../hooks/use-connect-site';
 import useMyJetpackConnection from '../../hooks/use-my-jetpack-connection';
-import { InfoTooltip } from '../info-tooltip';
 import cloud from './cloud.svg';
+import { ConnectionOwnerInfo } from './connection-owner-info';
 import emptyAvatar from './empty-avatar.svg';
 import jetpackGray from './jetpack-gray.svg';
 import jetpack from './jetpack.svg';
 import styles from './styles.module.scss';
-import type {
-	ConnectionListItemType,
-	getSiteConnectionLineDataType,
-	getUserConnectionLineDataType,
-	ConnectionStatusCardType,
-	ConnectionItemButtonType,
-} from './types';
+import { useConnectionState } from './use-connection-state';
+import type { ConnectionStatusCardType } from './types';
 import type { MouseEvent } from 'react';
 
-interface AccountError {
-	type: string;
-	message: string;
-	details?: Record< string, unknown >;
-}
-
-const ConnectionListItem: ConnectionListItemType = ( {
-	text,
-	actionText,
-	onClick,
-	status = 'success',
-} ) => {
-	let icon = check;
-	let statusStyles = '';
-
-	if ( status === 'info' ) {
-		icon = null;
-		statusStyles = '';
-	}
-
-	if ( status === 'success' ) {
-		icon = check;
-		statusStyles = styles.success;
-	}
-
-	if ( status === 'error' ) {
-		icon = info;
-		statusStyles = styles.error;
-	}
-
-	if ( status === 'warning' ) {
-		icon = info;
-		statusStyles = styles.warning;
-	}
-
-	if ( status === 'unlock' ) {
-		icon = lockOutline;
-		statusStyles = '';
-	}
-
-	return (
-		<div className={ styles[ 'list-item' ] }>
-			<Text className={ clsx( styles[ 'list-item-text' ], statusStyles ) }>
-				{ icon && <Icon icon={ icon } /> }
-				<span>{ text }</span>
-			</Text>
-			{ actionText && status !== 'success' && (
-				<ConnectionItemButton actionText={ actionText } onClick={ onClick } />
-			) }
-		</div>
-	);
-};
-
-const ConnectionItemButton: ConnectionItemButtonType = ( { actionText, onClick } ) => {
-	return (
-		<Button variant="link" weight="regular" onClick={ onClick }>
-			{ actionText }
-		</Button>
-	);
-};
-
-const getSiteConnectionLineData: getSiteConnectionLineDataType = ( {
```

Comment:
From what I can see, you moved this function to a separate component. Did you move all of the removed block to other components?

Reply:
I have not moved it, I rather removed it as the things have been simplified.